### DISCUSSION TITLE
how to switch window/pane focus?

### DISCUSSION THREAD

**Discussion by quickfold** [freeplane/freeplane#2051] [Help]
When I have two map windows/panes open side-by-side (within one Freeplane instance), I change focus by clicking in one or the other. Is there any command or keyboard shortcut that would allow me to switch focus between the two windows without using the mouse?

**Comment by euu2021** [comment]
If you have only two map views opened, this script solves the problem:

```
import org.freeplane.features.mode.Controller

Controller.getCurrentController().getMapViewManager().nextMapView()
```

If you have more than 2 map views, then we need to come up something more complex.

**Reply by quickfold** [reply]
@euu2021 , thanks! I haven't used scripts in this way. A few questions:

1. How do I run the script?
2. Do I need to run it every time I want to use it, or is there a way to run it automatically?
3. How can I map the script to a hotkey?

**Reply by euu2021** [reply]
This is just a usual script. 

To run:
- go into Tools->Open user directory
- open scripts folder
- create a txt file with the script as content
- rename to file to end with `.groovy`
- restart FP
- find it under Tools->Scripts

Assign a hotkey the usual way. For example: Tools->Assign hot key

> Do I need to run it every time I want to use it, or is there a way to run it automatically?

Each time you run the script is like pressing a button that alternates the map views.

**Reply by quickfold** [reply]
Thank you! When I run the script, I get this error:

![image](https://github.com/user-attachments/assets/044bfe50-04fd-4178-a97a-d58c02b7ba90)

**Reply by euu2021** [reply]
Strange 🤔

Try creating a new script from `Tools->Edit script…`. It should work like this:

![chrome_DRzCN9TvFg](https://github.com/user-attachments/assets/86a365e5-2eb3-4753-a87a-7ca22b1258b8)

**Reply by i-plasm** [reply]
I wonder if you have scripts disabled in your preferences. You can check it here: `Tools->Preferences…`, go to the section `Plugins > Scripting` (the `Script execution enabled` option should say 'yes' or 'ask')

**Reply by quickfold** [reply]
@i-plasm , you were right! @euu2021 , it works perfectly, many thanks!
